x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/rule_migrations_task_client.ts

```diff
@@ -115,44 +124,63 @@ export class RuleMigrationsTaskClient {
 
   /** Returns the stats of a migration */
   public async getStats(migrationId: string): Promise<RuleMigrationTaskStats> {
+    const migration = await this.data.migrations.get({ id: migrationId });
+    if (!migration) {
+      throw new Error(`Migration with ID ${migrationId} not found`);
+    }
     const dataStats = await this.data.rules.getStats(migrationId);
-    const taskStats = this.getTaskStats(migrationId, dataStats.rules);
+    const taskStats = this.getTaskStats(migration, dataStats.rules);
     return { ...taskStats, ...dataStats };
   }
 
   /** Returns the stats of all migrations */
   async getAllStats(): Promise<RuleMigrationTaskStats[]> {
     const allDataStats = await this.data.rules.getAllStats();
-    return allDataStats.map((dataStats) => {
-      const taskStats = this.getTaskStats(dataStats.id, dataStats.rules);
-      return { ...taskStats, ...dataStats };
-    });
+    const allMigrations = await this.data.migrations.getAll();
+    const allMigrationsMap = new Map<string, StoredSiemMigration>(
+      allMigrations.map((migration) => [migration.id, migration])
+    );
+
+    const allStats: RuleMigrationTaskStats[] = [];
+
+    for (const dataStats of allDataStats) {
+      const migration = allMigrationsMap.get(dataStats.id);
+      if (migration) {
+        const taksStats = this.getTaskStats(migration, dataStats.rules);
+        allStats.push({ ...taksStats, ...dataStats });
+      }
+    }
+    return allStats;
   }
 
   private getTaskStats(
-    migrationId: string,
+    migration: StoredSiemMigration,
     dataStats: RuleMigrationDataStats['rules']
   ): Pick<RuleMigrationTaskStats, 'status' | 'last_error'> {
-    const lastError = RuleMigrationsTaskClient.migrationsLastError.get(migrationId);
+    const lastError = migration?.last_execution?.error;
     return {
-      status: this.getTaskStatus(migrationId, dataStats),
-      ...(lastError && { last_error: lastError.message }),
+      status: this.getTaskStatus(migration, dataStats),
+      ...(lastError && { last_error: lastError }),
     };
   }
 
   private getTaskStatus(
-    migrationId: string,
+    migration: StoredSiemMigration,
     dataStats: RuleMigrationDataStats['rules']
   ): SiemMigrationTaskStatus {
+    const { id: migrationId, last_execution: lastExecution } = migration;
     if (this.migrationsRunning.has(migrationId)) {
       return SiemMigrationTaskStatus.RUNNING;
     }
-    if (dataStats.pending === dataStats.total) {
-      return SiemMigrationTaskStatus.READY;
-    }
     if (dataStats.completed + dataStats.failed === dataStats.total) {
       return SiemMigrationTaskStatus.FINISHED;
     }
+    if (lastExecution?.is_aborted) {
+      return SiemMigrationTaskStatus.ABORTED;
+    }
+    if (dataStats.pending === dataStats.total) {
```

Comment:
Even if `pending === total` but the task was aborted, we would want to inform user that the task was aborted but can still be started.